src/vcsparser.core/CosmosDbOutputProcessor.cs

```diff
@@ -1,65 +1,111 @@
 
-            logger.LogToConsole($"Deleted: {documentsDeleted} existing documents of '{GetDocumentType<T>()}' type");
+            var listOfFilesPerDay = dataConverter.ConvertDictToOrderedListPerDay(dict);
+            var cosmosDocumentsToInsert = (from x in listOfFilesPerDay from valueValue in x.Value.Values select ConvertOutputJsonToCosmosDataDocument(valueValue, DocumentTypeHelper.GetDocumentType<T>(), x.Key)).ToList();
 
-            foreach (var document in listOfFilesPerDay)
-                dataDocumentRepository.CreateDataDocument(document);
+            if (cosmosDocumentsToInsert.Count >= batchSize)
+            {
+                logger.LogToConsole("Inserting documents with batches.");
+
+                BatchDeleteDocuments<T>(dict.First().Key, dict.Last().Key);
+                BatchInsertDocuments(cosmosDocumentsToInsert);
+            }
+            else
+            {
+                DeleteDocuments(cosmosDocumentsToInsert);
+                InsertDocuments(cosmosDocumentsToInsert);
+            }
+
+            logger.LogToConsole($"Finished inserting documents to cosmos for ('{projectName}') project of '({DocumentTypeHelper.GetDocumentType<T>()}') type.");
         }
 
         public Dictionary<DateTime, Dictionary<string, T>> GetDocumentsInDateRange<T>(DateTime fromDateTime, DateTime endDateTime) where T : IOutputJson
         {
-            var documents = dataDocumentRepository.GetDocumentsInDateRange<T>(projectName, GetDocumentType<T>(), fromDateTime, endDateTime);
+            logger.LogToConsole($"Requesting for ('{projectName}') documents of '({DocumentTypeHelper.GetDocumentType<T>()}') type in date range: {fromDateTime} to {endDateTime}.");
+
+            var documents = dataDocumentRepository.GetDocumentsInDateRange<T>(projectName, DocumentTypeHelper.GetDocumentType<T>(), fromDateTime, endDateTime);
+            var data = ConvertDataFromCosmosDb(documents);
+
+            logger.LogToConsole($"Received {data.Count} ('{projectName}') documents of '({DocumentTypeHelper.GetDocumentType<T>()}') type in date range: {fromDateTime} to {endDateTime}.");
+
+            return data;
+        }
+
+        public Dictionary<DateTime, Dictionary<string, T>> GetAllDocumentsByProjectNameAndDocumentType<T>() where T : IOutputJson
+        {
+            logger.LogToConsole($"Requesting for all ('{projectName}') documents of '({DocumentTypeHelper.GetDocumentType<T>()}') type.");
+
+            var documents = dataDocumentRepository.GetAllDocumentsByProjectAndDocumentType<T>(projectName, DocumentTypeHelper.GetDocumentType<T>());
+            var data = ConvertDataFromCosmosDb(documents);
+
+            logger.LogToConsole($"Received {data.Count} ('{projectName}') documents of '({DocumentTypeHelper.GetDocumentType<T>()}') type.");
+
+            return data;
+        }
+
+        private Dictionary<DateTime, Dictionary<string, T>> ConvertDataFromCosmosDb<T>(List<CosmosDataDocument<T>> documents) where T : IOutputJson
```

Comment:
This function is getting a bit too chunky. Can you please split in smaller ones?

Reply:
Done.